libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/internal/route/line/MapboxRouteLineUtils.kt

```diff
@@ -1069,11 +1069,19 @@ internal object MapboxRouteLineUtils {
         useLineMetrics: Boolean
     ) {
         if (!style.styleSourceExists(layerSource)) {
-            geoJsonSource(layerSource) {
-                maxzoom(16)
-                lineMetrics(useLineMetrics)
-                tolerance(tolerance)
-            }.bindTo(style)
+            style.addStyleSource(
+                layerSource,
+                Value(
+                    hashMapOf(
+                        "type" to Value("geojson"),
+                        "data" to Value(""),
+                        "sharedCache" to Value(true),
```

Comment:
Also capturing from the chat, let's make this disabled by default and expose a new option in `MapboxRouteLineOptions` annotated with `@ExperimentalPreviewMapboxNavigationAPI`.